src/Symfony/Component/Messenger/EventListener/SendFailedMessageForRetryListener.php

```diff
@@ -68,6 +72,12 @@ public function onMessageFailed(WorkerMessageFailedEvent $event)
 
             // re-send the message for retry
             $this->getSenderForTransport($event->getReceiverName())->send($retryEnvelope);
+
+            if (null === $this->eventDispatcher) {
```

Comment:
please inverse the logic here `if (null !== $this->eventDispatcher) {`. no point in early return here.